apps/system/js/bluetooth_v2.js

```diff
@@ -0,0 +1,367 @@
+      this._setProfileConnected(this.Profiles.A2DP, evt.status);
+    });
+
+    adapter.addEventListener('scostatuschanged', (evt) => {
+      this._setProfileConnected(this.Profiles.SCO, evt.status);
+    });
+  },
+
+  /**
+   * BT APIv2: Watch 'onattributechanged' event from
+   * mozBluetooth.defaultAdapter for updating state information.
+   *
+   * @private
+   * @param  {Object} evt event object
+   */
+  _bluetoothAdapterAttrChangeHandler:
+    function bt__bluetoothAdapterAttrChangeHandler(evt) {
+      for (var i in evt.attrs) {
+        switch (evt.attrs[i]) {
+          case 'state':
+            if (this._adapter.state === 'enabled') {
+              this._isEnabled = true;
+              window.dispatchEvent(new CustomEvent('bluetooth-enabled'));
+              // this.icon && this.icon.update();
+            } else if (this._adapter.state === 'disabled') {
+              this._isEnabled = false;
+              window.dispatchEvent(new CustomEvent('bluetooth-disabled'));
+              // this.icon && this.icon.update();
+            }
+            break;
+          default:
+            break;
+        }
+      }
+  },
+
+  /**
+   * Watch 'onattributechanged' event from mozBluetooth for updating default
+   * adapter information.
+   *
+   * 'onattributechanged' event description:
+   * A handler to trigger when bluetooth manager's only property
+   * defaultAdapter has changed.
+   *
+   * @private
+   * @param  {Object} evt event object
+   */
+  _bluetoothAttrChangeHandler: function bt__bluetoothAttrChangeHandler(evt) {
+    for (var i in evt.attrs) {
+      switch (evt.attrs[i]) {
+        case 'defaultAdapter':
+          // Default adapter attribute change.
+          // Usually, it means that we reach new default adapter.
+          this._adapter = this._bluetooth.defaultAdapter;
+          this._adapter.addEventListener('attributechanged',
+            this._bluetoothAdapterAttrChangeHandler.bind(this));
+          this._updateProfileStat(this._adapter);
+          this.debug('defaultAdapter changed.');
+          Promise.resolve(this._adapter);
+          break;
+        default:
+          break;
+      }
+    }
+  },
+
+  /**
+   * Called by external for re-use adapter.
+   *
+   * @public
+   */
+  getAdapter: function bt_getAdapter() {
+    return new Promise(function(resolve, reject) {
+      if (this._adapter) {
+        this.debug('return cached adapter');
+        resolve(this._adapter);
+      } else {
+        this.debug('try to get adapter');
+        // need time to get bluetooth adapter at first run
+        this._bluetooth.addEventListener('attributechanged',
+          this._attributeHandler);
+      }
+    }.bind(this));
+  },
+
+  /**
+   * Maintain bluetooth enable/disable stat.
+   *
+   * @public
+   */
+  get isEnabled() {
+    return this._isEnabled;
+  },
+
+  /**
+   * Console log.
+   *
+   * @param  {[type]} msg debug message
+   */
+  debug: function bt_debug(msg) {
```

Comment:
mmm... `_debug` has been used for flag. I could use `_onDebug` as flag instead as keyboard_manager does